### PR TITLE
4156: Fix missing booking url in app feed for p2b events

### PR DESCRIPTION
**Pull request by holt83**
#### Link to issue

https://platform.dandigbib.org/issues/4156

#### Description

In the old version of the place2book module it hooked into the existing ticket link field on ding_event. When node was saved/updated it ensured that the booking URL from place2book was inserted into the ticket_link field. When node was shown it would modify the output by the ticket link field to show the correct p2b-status.

When making the migration module for the new p2b integration, [we decided to remove the p2b-URL from the ticket link field](https://github.com/ding2/ding2/blob/7.x-4.7.0-rc1/modules/ding_place2book/modules/ding_place2book_migrate/ding_place2book_migrate.module#L78). This was because the new p2b module defined it's own field with a formatter that outputs it's own book button and keeping the p2b-URL in the ticket link field as is, would result in double book buttons in frontend.

Removing the pb2-URL from ticket link has had the unfortunate effect that the booking URL from p2b wasn't included in the RSS feed used by the app, since this RSS view just pulls from that field.

The easiest approach seems to be to revert back to old ways of doing things and keep the ticket link field synced with p2b-URL and modify the ticket link output. We do this by using a hidden formatter instead and implement hook_process_node().

Since no libraries have done the migration in production yet, we can just remove the code that removed the p2b-URL during migration.

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

### CODE REVIEW

**Comment by guddo** [comment]
@holt83 Did you test this with the migration? So like from before migration to after this change-set?

**Comment by holt83** [comment]
@guddo Yes I tested with migration. We just keep the URL in ticket link field now.

File: modules/ding_place2book/ding_place2book.module [hunk 1]
```diff
@@ -257,6 +358,39 @@ function ding_place2book_instance() {
...
+  }
+
+  $field_place2book = $field_place2book[0];
+  if (empty($field_place2book['synchronize'])) {
+    return FALSE;
+  }
+  // The event is synchronized and this is just a safety check to ensure these
+  // important values is present and non-empty.
+  if (empty($field_place2book['event_id']) || empty($field_place2book['event_maker_id'])) {
+    return FALSE;
```
[the remark this answers is not in the document]
Comment (holt83) [reply modules/ding_place2book/ding_place2book.module]: Good idea :) I'll update the PR with your suggestion.

**Comment by holt83** [comment]
@cableman I updated PR with you suggestion to log in watchdog.

Also sneaked in a fix for some notices: https://github.com/ding2/ding2/pull/1382/commits/e037f74fa579cf4ab0f92dfebdd6dcfb874918c4

**Pull request merged**